### ISSUE TITLE
Add support for frame instances & identical frame re-use

### ISSUE THREAD

**Issue by jsmars** [aseprite/aseprite#2170]
### Non-linear instances
It would be very valuable to be able to duplicate frames as instances of the original frame, so that editing an instance will always copy over immediately to the other instances of the same frame. Similarly to what linked cells currently does, but without having them forced to be one after another.

Additionally, being able to move/rotate an instanced layer without editing the pixel locations would also be very valuable. That way an object that stays the same, but moves around in an animation, can be edited from one frame.

The main benefit of this would be greatly increased efficient workflow for working with spritesheets for games where often parts will be reused over and over again, but not necessarily always in the order of the frames.

### Frame re-use
Finally an additional huge benefit of supporting this and working with large spritesheets, and partly a feature request on it's own, would be when exporting animations, if you can detect complete frames that have identical pixels, that the frame will be reused on the spritesheet and thus optimize a lot of texture space for animations where some frames may appear multiple times, for example in blinking UI elements.

**Comment by dacap** [comment]
Hi @jsmars, actually you can put linked cel far away:

<img width="555" alt="Screen Shot 2019-10-09 at 18 33 17" src="https://user-images.githubusercontent.com/39654/66522213-48a7f580-eac3-11e9-8dfe-2ce4aa587d3e.png">

About the instances of layers, it's related to #593 (quite complex feature).

About frame re-use, the next version will have some changes to improve the packaging of sprite sheets and re-use similar frames in the same location if that is needed.

**Comment by jsmars** [comment]
Yeah the linked cells are a very nice feature, but if I would like to be able to be able to have them linked even if they are not all in a row (ie. two different links, but connected). For example if I create a character sprite sheet with 10 different animations, number 1, 3, and 7 may be using the same body that could be linked, but I can't manage to have them in a row in the timeline and at the same time have animations 2, 4 and 5 that contain a different cell that is linked.

Really looking forward to the re-use in the next version too. Thanks for the good work @dacap ! :)

**Comment by zhengying** [comment]
This feature must be a killer feature, Any plans?
